Tasks/AzureFunctionAppV2/operations/AzureAppServiceUtility.ts

```diff
@@ -409,18 +418,41 @@ export class AzureAppServiceUtility {
             if (pe && pe.value && pe.value.length && pe.value.length > 0){
                 tl.debug("Function app has Private Endpoints.");
                 return "true";
+                }
+                else {            
+                    tl.debug("Function app has NO Private Endpoints.");
+                    return "false";
+                }
             }
-            else {            
-                tl.debug("Function app has NO Private Endpoints.");
-                return "false";
+            catch(error){
+                tl.debug(`Skipping private endpoint check: ${error}`);
+                return null;
+            
+        }      
+        
+    }              
+    
+
+    public async isSitePublishingCredentialsEnabled(): Promise<boolean>{
+        try{
+            let scmAuthPolicy: any = await this._appService.getSitePublishingCredentialPolicies();
+            tl.debug(`Site Publishing Policy check: ${JSON.stringify(scmAuthPolicy)}`);
+            if(scmAuthPolicy && scmAuthPolicy.properties.allow) {
+                tl.debug("Function App does allow SCM access");
+                return true;
+            }
+            else {
+                tl.debug("Function App does not allow SCM Access");
+                return false;
             }
         }
         catch(error){
-            tl.debug(`Skipping private endpoint check: ${error}`);
+            tl.debug(`Skipping SCM Policy check: ${error}`);
```

Comment:
Should we fallback to basic auth (old path) if this call fails for some reason ? would that be more safe from compat perspective ? Or do you think SPN always have access to get credentials ? 

Could there be a case where SPN permissions are modified since it was created ? i.e it was able to access publish profile during pipeline setup but not able to anymore during pipeline run.